app/lib/pre_assembly/file_manifest.rb

```diff
@@ -6,37 +6,55 @@ module PreAssembly
 
   # It is used by pre-assembly during the accessioning process to produce custom content metadata if a file manifest is supplied
   class FileManifest
-    attr_reader :manifest, :csv_filename, :staging_location
+    attr_reader :csv_filename, :staging_location
 
     # the valid roles a file can have, if you specify a "role" column and the value is not one of these, it will be ignored
     VALID_ROLES = %w[transcription annotations derivative master].freeze
 
     def initialize(staging_location:, csv_filename:)
       @staging_location = staging_location
       @csv_filename = csv_filename
-      # read CSV
-      @manifest = load_manifest # this will cache the entire file manifest csv in @manifest
     end
 
     def exists?
       File.exist?(csv_filename)
     end
 
     # rubocop:disable Metrics/AbcSize
-    def load_manifest
-      # load file into @rows and then build up @manifest
-      rows = CsvImporter.parse_to_hash(@csv_filename)
-      sequence = nil
-      rows.each_with_object({}) do |row, manifest|
-        object = row[:object]
-        sequence = row[:sequence].to_i if row[:sequence].present?
-        manifest[object] ||= { file_sets: {} }
-        manifest[object][:file_sets][sequence] ||= file_set_properties_from_row(row, sequence)
-        manifest[object][:file_sets][sequence][:files] << file_properties_from_row(row)
+    def manifest
+      @manifest ||= begin
```

Comment:
I guess this `||=` is more similar to how we do things now. It seems less obvious though, and there's no way to reload the manifest anymore, which was possible before (but isn't done anywhere).

Reply:
Part of what I was trying to do with this refactor was to clarify the public interface of this class, as I found it hard to understand what this class was actually doing. This led me to realize that `load_manifest` was only being used to load the `manifest` instance variable. And the idiomatic way to do this is `||=` rather than set a class instance variable in the initializer.

(I don't consider the `||=` as crucial as clarifying the public interface.)